pkg/vm/engine/tae/tables/txnentries/mergeobjects.go

```diff
@@ -106,14 +113,34 @@ func (entry *mergeObjectsEntry) prepareTransferPage() {
 			id := obj.AsCommonID()
 			id.SetBlockOffset(uint16(j))
 			page := model.NewTransferHashPage(id, time.Now(), len(m), isTransient)
+			mapping := make(map[uint32][]byte, len(m))
 			for srcRow, dst := range m {
 				objID := entry.createdObjs[dst.ObjIdx].ID()
 				blkID := objectio.NewBlockidWithObjectID(objID, uint16(dst.BlkIdx))
-				page.Train(uint32(srcRow), *objectio.NewRowid(blkID, uint32(dst.RowIdx)))
+				rowID := objectio.NewRowid(blkID, uint32(dst.RowIdx))
+				mapping[uint32(srcRow)] = rowID[:]
 			}
+			page.Train(mapping)
+
+			start = time.Now()
+			err := AddTransferPage(page, ioVector)
+			if err != nil {
+				return
+			}
+			duration += time.Since(start)
+
 			entry.pageIds = append(entry.pageIds, id)
 			_ = entry.rt.TransferTable.AddPage(page)
+			pages = append(pages, page)
 		}
+
+		start = time.Now()
+		err := WriteTransferPage(pages, *ioVector)
+		if err != nil {
+			return
```

Comment:
Any error, return ok? i think we need to inject some error to see whether we handle error as expected. (memory or storage leak, panic, or something horrible)